src/border_agent/border_agent.cpp

```diff
@@ -165,6 +167,28 @@ BorderAgent::BorderAgent(otbr::Ncp::RcpHost &aHost, Mdns::Publisher &aPublisher)
     mHost.AddThreadStateChangedCallback([this](otChangedFlags aFlags) { HandleThreadStateChanged(aFlags); });
 }
 
+otbrError BorderAgent::CreateEphemeralKey(std::string &aEphemeralKey)
+{
+    std::random_device              r;
+    std::default_random_engine      engine(r());
+    std::uniform_int_distribution<> dist('0', '9'); // ASCII code for the digits.
```

Comment:
Is this random-generator suited for crypto use-cases?
I think better to use crypto-safe one here. Can consider using the OT public API `otRandomCryptoFillBuffer()` for this?

Reply:
Thanks for the suggestion! Have modified accordingly to use the crypto random now.